main/src/components/docs/LegendItemWithSplit.jsx

```diff
@@ -0,0 +1,201 @@
+import React, { useEffect, useState, useRef } from 'react';
+import styles from './LegendItem.module.css';
+import SvgMeasurementComponent from './SvgMeasurementComponent.jsx';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+const svgCache = new Map();
+export const TOP_LEVEL_GROUPS_SELECT = ':scope > g';
+
+export default function LegendItemWithSplit({ svgPath, svgParts }) {
+
+  const SVG_DAY_FILE_SUFFIX = '-day.svg';
+  const SVG_NIGHT_FILE_SUFFIX = '-night.svg';
+  const WIDTH_ATTR = 'width';
+  const HEIGHT_ATTR = 'height';
+  const VIEW_BOX_ATTR = 'viewBox';
+  const TRANSFORM_ATTR = 'transform';
+  const DEFS_SELECT = 'defs';
+  const STYLE_SELECT = 'style';
+  const [splitSvgsDay, setSplitSvgsDay] = useState([]);
+  const [splitSvgsNight, setSplitSvgsNight] = useState([]);
+  const [svgContentDay, setSvgContentDay] = useState('');
+  const [svgContentNight, setSvgContentNight] = useState('');
+  const [loadingDay, setLoadingDay] = useState(true);
+  const [loadingNight, setLoadingNight] = useState(true);
+  const COLUMN_COUNT = 3;
+  const measurementComponentRef = useRef(null);
+  const DEBUG = false;
+
+  const useSvgContent = ((svgPath, setSvgContent, setLoading, setSplitSvgs) => {
+    if (!svgPath) {
+      setSvgContent('');
+      setLoading(false);
+      return;
+    }
+    setLoading(true)
+    if (svgCache.has(svgPath)) {
+      setSplitSvgs(svgCache.get(svgPath));
+      return;
+    }
+    fetch(svgPath)
+      .then(response => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        return response.text();
+      })
+      .then(data => {
+        setSvgContent({ svgPath, data });
+        setLoading(false);
+      })
+      .catch(err => {
+        DEBUG && console.error('Error fetching SVG:', err);
+        setLoading(false);
+      });
+  })
+
+  useEffect(() => {
+    useSvgContent(svgPath + SVG_DAY_FILE_SUFFIX, setSvgContentDay, setLoadingDay, setSplitSvgsDay);
+    useSvgContent(svgPath + SVG_DAY_FILE_SUFFIX, setSvgContentNight, setLoadingNight, setSplitSvgsNight);
+  }, [svgPath]);
+
+  function splitSvgToArray(svgContent) {
```

Comment:
where is useMemo?

Reply:
If we use cache, it is calculated only once.